### PR TITLE
#5389: Move ttnn.repeat to c++

### PR DESCRIPTION
**Pull request by ayerofieiev-tt**
**Whats happening**
This change is a part of the #5389

### CODE REVIEW

File: ttnn/cpp/pybind11/operations/data_movement.hpp [hunk 1]
```diff
@@ -75,11 +79,34 @@ The algorithms available for upsampling are 'nearest' for now.
...
+
+    >>> tensor = ttnn.repeat(ttnn.from_torch(torch.tensor([[1, 2], [3, 4]]), 2,)), device)
+    >>> print(tensor)
+    tensor([[1, 2],
+    [1, 2],
+    [3, 4],
+    [3, 4]])
+        )doc",
+        ttnn::pybind_arguments_t{
+            py::arg("input_tensor"), py::arg("shape"), py::kw_only(), py::arg("memory_config") = std::nullopt});
```
Comment (TT-BrianLiu) [inline ttnn/cpp/pybind11/operations/data_movement.hpp:109]: Indentation?

File: ttnn/cpp/ttnn/operations/data_movement.hpp [hunk 2]
```diff
@@ -254,7 +256,36 @@ struct UpSample {
     }
 };
 
+struct Repeat {
+    static inline const std::array<TensorSchema, 1> input_tensor_schemas() {
+        return {ttnn::TensorSchema{
+            4,
+            4,
```
Comment (TT-BrianLiu) [inline ttnn/cpp/ttnn/operations/data_movement.hpp:263]: are these tensor ranks? Maybe leave a comment describing

**Review by TT-BrianLiu — APPROVED** [review]
Minor comments

File: ttnn/cpp/pybind11/operations/data_movement.hpp (hunk 1 above)
Comment (ayerofieiev-tt) [reply ttnn/cpp/pybind11/operations/data_movement.hpp:109]: clang format

Comment (ayerofieiev-tt) [reply ttnn/cpp/pybind11/operations/data_movement.hpp:109]: My bad, had to open as a draft.
Can't get it to run properly and have a few other minor corrections.

File: ttnn/cpp/pybind11/operations/data_movement.hpp [hunk 3]
```diff
@@ -14,8 +15,9 @@ namespace ttnn {
 namespace operations {
 namespace data_movement {
 void py_module(py::module& module) {
-
-    module.def("permute", &permute,
+    module.def(
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/pybind11/operations/data_movement.hpp:18]: Format changes are due to clang-format not being run on the whole codebase yet.

**Review by arakhmati — APPROVED** [review]

**Review by arakhmati — APPROVED** [review]

**Pull request merged**